snort3: try to get  vectorscan and gperftools auto

graysky2 opened the pull request:
Trying to autoselect based on simply selecting the snort3 package. Right now, only gperftools-runtime works.

**Package details**

    Maintainer: @\<github-user> (find it by checking the history of the package Makefile)
    Description:

**Indication of run testing**

     OpenWrt version:
     OpenWrt target/subtarget:
     OpenWrt device:

**Formalities**

     [ ] Review the [CONTRIBUTING.md](../CONTRIBUTING.md) file for detailed contributing guidelines.

     If your PR contains a patch
     [ ] Make sure that it can be applied by `git am`
     [ ] It must be refreshed to avoid offsets, fuzzes, etc by `make package/foo/refresh V=s`
     [ ] It must be in a way that it is potentially upstreamable (subject, commit description, etc.), and we must try to upstream it so we have fewer patches and fewer.

BKPepe committed:
snort3: refactor architecture-specific dependencies and CMake options

1. Enabled hyperscan/vectorscan together with adding dependency only for x86_64 and aarch64.
2. Disabled tmalloc (from gperftools package) for powerpc and mips.

By doing this refactor, snort3 is going to be available for more OpenWrt devices
(as it was in the past) as currently it was compiled only for x86_x64 and aarch64 by mistake.

Fixes: 257e2fc38aa694604e7be9e70e58082037133434 ("snort3: fix logic in gpertools-runtime depends")

Build system: x86/64
Build-tested: x86/64-glibc
Run-tested: x86/64-glibc

Signed-off-by: Josef Schlehofer <pepe.schlehofer@gmail.com>
Co-authored-by: John Audia <therealgraysky@proton.me>
Signed-off-by: John Audia <therealgraysky@proton.me>

graysky2 commented:
Do you have any thoughts as to why this strategy will not select vectorscan? It does select gperftools at least with x86_64.

BKPepe commented:
I tried similar thing and GitHub Actions said it was no go.

graysky2 commented:
There has to be some assumption we are making that the build system is invalidating. That fact that it works to select gperftools is proof the strategy is sound. I thought using two differently named variables would solve it but alas no.

Here is the output of the [script](https://gist.github.com/graysky2/c77c9d063d4ade90f079e4fcd1b64080) with the change in this PR. You can see the step forward it makes for gperftools.
<details>
<summary>Click here</summary>
| Target_Subtarget | Snort3 | Gperftools | Vectorscan |
|------------------|--------|------------|------------|
| airoha_an7581    | ✔    | ✔        | ✘        |
| airoha_an7583    | ✔    | ✔        | ✘        |
| airoha_en7523    | ✔    | ✔        |            |
| apm821xx_nand    |        |            |            |
| apm821xx_sata    |        |            |            |
| at91_sam9x       | ✔    | ✔        |            |
| at91_sama5       | ✔    | ✔        |            |
| at91_sama7       | ✔    | ✔        |            |
| ath79_generic    |        |            |            |
| ath79_mikrotik   |        |            |            |
| ath79_nand       |        |            |            |
| ath79_tiny       |        |            |            |
| bcm27xx_bcm2708  | ✔    | ✔        |            |
| bcm27xx_bcm2709  | ✔    | ✔        |            |
| bcm27xx_bcm2710  | ✔    | ✔        | ✘        |
| bcm27xx_bcm2711  | ✔    | ✔        | ✘        |
| bcm27xx_bcm2712  | ✔    | ✔        | ✘        |
| bcm47xx_generic  |        |            |            |
| bcm47xx_legacy   |        |            |            |
| bcm47xx_mips74k  |        |            |            |
| bcm4908_generic  | ✔    | ✔        | ✘        |
| bcm53xx_generic  | ✔    | ✔        |            |
| bmips_bcm6318    |        |            |            |
| bmips_bcm63268   |        |            |            |
| bmips_bcm6328    |        |            |            |
| bmips_bcm6358    |        |            |            |
| bmips_bcm6362    |        |            |            |
| bmips_bcm6368    |        |            |            |
| d1_generic       |        | ✘        |            |
| econet_en751221  |        |            |            |
| gemini_generic   | ✔    | ✔        |            |
| imx_cortexa53    | ✔    | ✔        | ✘        |
| imx_cortexa7     | ✔    | ✔        |            |
| imx_cortexa9     | ✔    | ✔        |            |
| ipq40xx_chromium | ✔    | ✔        |            |
| ipq40xx_generic  | ✔    | ✔        |            |
| ipq40xx_mikrotik | ✔    | ✔        |            |
| ipq806x_chromium | ✔    | ✔        |            |
| ipq806x_generic  | ✔    | ✔        |            |
| ixp4xx_generic   | ✔    | ✔        |            |
| kirkwood_generic | ✔    | ✔        |            |
| lantiq_ase       |        |            |            |
| lantiq_falcon    |        |            |            |
| lantiq_xrx200    |        |            |            |
| lantiq_xrx200_legacy |        |            |            |
| lantiq_xway      |        |            |            |
| lantiq_xway_legacy |        |            |            |
| layerscape_armv7 | ✔    | ✔        |            |
| layerscape_armv8_64b | ✔    | ✔        | ✘        |
| loongarch64_generic |        | ✘        |            |
| malta_be         |        |            |            |
| malta_be64       |        |            |            |
| malta_le         |        |            |            |
| malta_le64       |        | ✘        |            |
| mediatek_filogic | ✔    | ✔        | ✘        |
| mediatek_mt7622  | ✔    | ✔        | ✘        |
| mediatek_mt7623  | ✔    | ✔        |            |
| mediatek_mt7629  | ✔    | ✔        |            |
| mpc85xx_p1010    |        |            |            |
| mpc85xx_p1020    |        |            |            |
| mpc85xx_p2020    |        |            |            |
| mvebu_cortexa53  | ✔    | ✔        | ✘        |
| mvebu_cortexa72  | ✔    | ✔        | ✘        |
| mvebu_cortexa9   | ✔    | ✔        |            |
| mxs_generic      | ✔    | ✔        |            |
| omap_generic     | ✔    | ✔        |            |
| pistachio_generic |        |            |            |
| qoriq_generic    |        | ✘        |            |
| qualcommax_ipq50xx | ✔    | ✔        | ✘        |
| qualcommax_ipq60xx | ✔    | ✔        | ✘        |
| qualcommax_ipq807x | ✔    | ✔        | ✘        |
| qualcommbe_ipq95xx | ✔    | ✔        | ✘        |
| ramips_mt7620    |        |            |            |
| ramips_mt7621    |        |            |            |
| ramips_mt76x8    |        |            |            |
| ramips_rt288x    |        |            |            |
| ramips_rt305x    |        |            |            |
| ramips_rt3883    |        |            |            |
| realtek_rtl838x  |        |            |            |
| realtek_rtl839x  |        |            |            |
| realtek_rtl930x  |        |            |            |
| realtek_rtl930x_nand |        |            |            |
| realtek_rtl931x  |        |            |            |
| realtek_rtl931x_nand |        |            |            |
| rockchip_armv8   | ✔    | ✔        | ✘        |
| sifiveu_generic  |        | ✘        |            |
| siflower_sf19a2890 |        |            |            |
| siflower_sf21    |        | ✘        |            |
| stm32_stm32mp1   | ✔    | ✔        |            |
| sunxi_cortexa53  | ✔    | ✔        | ✘        |
| sunxi_cortexa7   | ✔    | ✔        |            |
| sunxi_cortexa8   | ✔    | ✔        |            |
| tegra_generic    | ✔    | ✔        |            |
| uml_generic      | ✔    | ✔        | ✘        |
| x86_64           | ✔    | ✔        | ✘        |
| x86_generic      | ✔    | ✔        |            |
| x86_geode        | ✔    | ✔        |            |
| x86_legacy       | ✔    | ✔        |            |
| zynq_generic     | ✔    | ✔        |            |

✔ means enabled ; ✘ means available but disabled ; blank means unavailable

</details>

BKPepe commented:
Hmm, based on the table, it is not working as it should, because for mpc85xx is not working.

BKPepe commented on net/snort3/Makefile:
```diff
@@ -21,15 +21,35 @@ PKG_CPE_ID:=cpe:/a:snort:snort
+  CMAKE_OPTIONS += \
+	-DENABLE_TCMALLOC=ON \
+	-DTCMALLOC_LIBRARIES=$(STAGING_DIR)/usr/lib/libtcmalloc.so
+  EXTRAGP:= +gperftools-runtime
```
I think this should be simplified to e.g.

1) DEPENDS+=
2) Package/snort3/DEPENDS += +gperftools-runtime

It should not be needed to create custom variable.

graysky2 replied:
Are you sure? I tried this and found that the buildsystem wasn't adding the options. I just pushed a new commit that I have verified working. Here is the output of the deptest.sh:

<details>
<summary>Click here</summary>

| Target_Subtarget | Snort3 | Gperftools | Vectorscan |
|------------------|--------|------------|------------|
| airoha_an7581    | ✔    | ✔        | ✔        |
| airoha_an7583    | ✔    | ✔        | ✔        |
| airoha_en7523    | ✔    | ✔        |            |
| apm821xx_nand    |        |            |            |
| apm821xx_sata    |        |            |            |
| at91_sam9x       | ✔    | ✔        |            |
| at91_sama5       | ✔    | ✔        |            |
| at91_sama7       | ✔    | ✔        |            |
| ath79_generic    |        |            |            |
| ath79_mikrotik   |        |            |            |
| ath79_nand       |        |            |            |
| ath79_tiny       |        |            |            |
| bcm27xx_bcm2708  | ✔    | ✔        |            |
| bcm27xx_bcm2709  | ✔    | ✔        |            |
| bcm27xx_bcm2710  | ✔    | ✔        | ✔        |
| bcm27xx_bcm2711  | ✔    | ✔        | ✔        |
| bcm27xx_bcm2712  | ✔    | ✔        | ✔        |
| bcm47xx_generic  |        |            |            |
| bcm47xx_legacy   |        |            |            |
| bcm47xx_mips74k  |        |            |            |
| bcm4908_generic  | ✔    | ✔        | ✔        |
| bcm53xx_generic  | ✔    | ✔        |            |
| bmips_bcm6318    |        |            |            |
| bmips_bcm63268   |        |            |            |
| bmips_bcm6328    |        |            |            |
| bmips_bcm6358    |        |            |            |
| bmips_bcm6362    |        |            |            |
| bmips_bcm6368    |        |            |            |
| d1_generic       |        | ✘        |            |
| econet_en751221  |        |            |            |
| gemini_generic   | ✔    | ✔        |            |
| imx_cortexa53    | ✔    | ✔        | ✔        |
| imx_cortexa7     | ✔    | ✔        |            |
| imx_cortexa9     | ✔    | ✔        |            |
| ipq40xx_chromium | ✔    | ✔        |            |
| ipq40xx_generic  | ✔    | ✔        |            |
| ipq40xx_mikrotik | ✔    | ✔        |            |
| ipq806x_chromium | ✔    | ✔        |            |
| ipq806x_generic  | ✔    | ✔        |            |
| ixp4xx_generic   | ✔    | ✔        |            |
| kirkwood_generic | ✔    | ✔        |            |
| lantiq_ase       |        |            |            |
| lantiq_falcon    |        |            |            |
| lantiq_xrx200    |        |            |            |
| lantiq_xrx200_legacy |        |            |            |
| lantiq_xway      |        |            |            |
| lantiq_xway_legacy |        |            |            |
| layerscape_armv7 | ✔    | ✔        |            |
| layerscape_armv8_64b | ✔    | ✔        | ✔        |
| loongarch64_generic |        | ✘        |            |
| malta_be         |        |            |            |
| malta_be64       |        |            |            |
| malta_le         |        |            |            |
| malta_le64       |        | ✘        |            |
| mediatek_filogic | ✔    | ✔        | ✔        |
| mediatek_mt7622  | ✔    | ✔        | ✔        |
| mediatek_mt7623  | ✔    | ✔        |            |
| mediatek_mt7629  | ✔    | ✔        |            |
| mpc85xx_p1010    |        |            |            |
| mpc85xx_p1020    |        |            |            |
| mpc85xx_p2020    |        |            |            |
| mvebu_cortexa53  | ✔    | ✔        | ✔        |
| mvebu_cortexa72  | ✔    | ✔        | ✔        |
| mvebu_cortexa9   | ✔    | ✔        |            |
| mxs_generic      | ✔    | ✔        |            |
| omap_generic     | ✔    | ✔        |            |
| pistachio_generic |        |            |            |
| qoriq_generic    |        | ✘        |            |
| qualcommax_ipq50xx | ✔    | ✔        | ✔        |
| qualcommax_ipq60xx | ✔    | ✔        | ✔        |
| qualcommax_ipq807x | ✔    | ✔        | ✔        |
| qualcommbe_ipq95xx | ✔    | ✔        | ✔        |
| ramips_mt7620    |        |            |            |
| ramips_mt7621    |        |            |            |
| ramips_mt76x8    |        |            |            |
| ramips_rt288x    |        |            |            |
| ramips_rt305x    |        |            |            |
| ramips_rt3883    |        |            |            |
| realtek_rtl838x  |        |            |            |
| realtek_rtl839x  |        |            |            |
| realtek_rtl930x  |        |            |            |
| realtek_rtl930x_nand |        |            |            |
| realtek_rtl931x  |        |            |            |
| realtek_rtl931x_nand |        |            |            |
| rockchip_armv8   | ✔    | ✔        | ✔        |
| sifiveu_generic  |        | ✘        |            |
| siflower_sf19a2890 |        |            |            |
| siflower_sf21    |        | ✘        |            |
| stm32_stm32mp1   | ✔    | ✔        |            |
| sunxi_cortexa53  | ✔    | ✔        | ✔        |
| sunxi_cortexa7   | ✔    | ✔        |            |
| sunxi_cortexa8   | ✔    | ✔        |            |
| tegra_generic    | ✔    | ✔        |            |
| uml_generic      | ✔    | ✔        | ✘        |
| x86_64           | ✔    | ✔        | ✔        |
| x86_generic      | ✔    | ✔        |            |
| x86_geode        | ✔    | ✔        |            |
| x86_legacy       | ✔    | ✔        |            |
| zynq_generic     | ✔    | ✔        |            |

</details>

BKPepe replied:
I would say so, it has the same function and it should be basically the same. However, your commit still needs a little bit more work. For many targets, there is missing snort3. I am mostly curious about mpc85xx_p2020 and because of that CI/CD in the packages repo will fail.

graysky2 replied:
I do not know about the mpc targets... just trying to help out on this.

BKPepe commented on net/snort3/Makefile:
```diff
@@ -21,15 +21,35 @@ PKG_CPE_ID:=cpe:/a:snort:snort
+  EXTRAGP:= +gperftools-runtime
+endif
+
+ifneq ($(findstring x86_64,$(CONFIG_ARCH))$(findstring aarch64,$(CONFIG_ARCH)),)
+  CMAKE_OPTIONS += \
+	-DENABLE_HYPERSCAN=ON \
+	-DHS_INCLUDE_DIRS=$(STAGING_DIR)/usr/include/hs
+  EXTRAVS:= +vectorscan-runtime
+else
```
I was thinking about variants e.g. ``snort3-full`` or rather ``snort3-custom``. That will :100: % work, but in my opinion it was like... this package is heavily customized for you and only you will be using it even it will be available for others. 🤷

graysky2 replied:
I agree. Let's not make it more complex.

graysky2 committed:
snort3: add conditional depends for gperftools and vectorscan

The original Makefile used Make conditionals (ifneq/findstring) to set
CMAKE_OPTIONS based on architecture, which worked correctly during the
build phase. However, the same approach fails for setting DEPENDS
because CONFIG_ARCH is not available when the dependency resolution
occurs during 'make defconfig'...

One solution is to add Config.in  with Kconfig-based conditional
logic that is evaluated by the config system at the correct time.
Two new config symbols are defined with architecture dependencies,
and these symbols are then used in the DEPENDS field to conditionally
pull in gperftools-runtime and vectorscan-runtime.

Checked using a script[1] which iterates over 99 subarches yielding
this result:

+-------------------------+--------+------------+------------+
| Target_Subtarget        | Snort3 | Gperftools | Vectorscan |
+-------------------------+--------+------------+------------+
| airoha_an7581           | ✔      | ✔          | ✔          |
| airoha_an7583           | ✔      | ✔          | ✔          |
| airoha_en7523           | ✔      | ✔          |            |
| apm821xx_nand           | ✔      |            |            |
| apm821xx_sata           | ✔      |            |            |
| at91_sam9x              | ✔      | ✔          |            |
| at91_sama5              | ✔      | ✔          |            |
| at91_sama7              | ✔      | ✔          |            |
| ath79_generic           | ✔      | ✘          |            |
| ath79_mikrotik          | ✔      | ✘          |            |
| ath79_nand              | ✔      | ✘          |            |
| ath79_tiny              | ✔      | ✘          |            |
| bcm27xx_bcm2708         | ✔      | ✔          |            |
| bcm27xx_bcm2709         | ✔      | ✔          |            |
| bcm27xx_bcm2710         | ✔      | ✔          | ✔          |
| bcm27xx_bcm2711         | ✔      | ✔          | ✔          |
| bcm27xx_bcm2712         | ✔      | ✔          | ✔          |
| bcm47xx_generic         | ✔      | ✘          |            |
| bcm47xx_legacy          | ✔      | ✘          |            |
| bcm47xx_mips74k         | ✔      | ✘          |            |
| bcm4908_generic         | ✔      | ✔          | ✔          |
| bcm53xx_generic         | ✔      | ✔          |            |
| bmips_bcm6318           | ✔      | ✘          |            |
| bmips_bcm63268          | ✔      | ✘          |            |
| bmips_bcm6328           | ✔      | ✘          |            |
| bmips_bcm6358           | ✔      | ✘          |            |
| bmips_bcm6362           | ✔      | ✘          |            |
| bmips_bcm6368           | ✔      | ✘          |            |
| d1_generic              |        | ✘          |            |
| econet_en751221         | ✔      | ✘          |            |
| gemini_generic          | ✔      | ✔          |            |
| imx_cortexa53           | ✔      | ✔          | ✔          |
| imx_cortexa7            | ✔      | ✔          |            |
| imx_cortexa9            | ✔      | ✔          |            |
| ipq40xx_chromium        | ✔      | ✔          |            |
| ipq40xx_generic         | ✔      | ✔          |            |
| ipq40xx_mikrotik        | ✔      | ✔          |            |
| ipq806x_chromium        | ✔      | ✔          |            |
| ipq806x_generic         | ✔      | ✔          |            |
| ixp4xx_generic          | ✔      | ✔          |            |
| kirkwood_generic        | ✔      | ✔          |            |
| lantiq_ase              | ✔      | ✘          |            |
| lantiq_falcon           | ✔      | ✘          |            |
| lantiq_xrx200           | ✔      | ✘          |            |
| lantiq_xrx200_legacy    | ✔      | ✘          |            |
| lantiq_xway             | ✔      | ✘          |            |
| lantiq_xway_legacy      | ✔      | ✘          |            |
| layerscape_armv7        | ✔      | ✔          |            |
| layerscape_armv8_64b    | ✔      | ✔          | ✔          |
| loongarch64_generic     |        | ✘          |            |
| malta_be                | ✔      | ✘          |            |
| malta_be64              | ✔      | ✘          |            |
| malta_le                | ✔      | ✘          |            |
| malta_le64              |        | ✘          |            |
| mediatek_filogic        | ✔      | ✔          | ✔          |
| mediatek_mt7622         | ✔      | ✔          | ✔          |
| mediatek_mt7623         | ✔      | ✔          |            |
| mediatek_mt7629         | ✔      | ✔          |            |
| mpc85xx_p1010           | ✔      |            |            |
| mpc85xx_p1020           | ✔      |            |            |
| mpc85xx_p2020           | ✔      |            |            |
| mvebu_cortexa53         | ✔      | ✔          | ✔          |
| mvebu_cortexa72         | ✔      | ✔          | ✔          |
| mvebu_cortexa9          | ✔      | ✔          |            |
| mxs_generic             | ✔      | ✔          |            |
| omap_generic            | ✔      | ✔          |            |
| pistachio_generic       | ✔      | ✘          |            |
| qoriq_generic           |        | ✘          |            |
| qualcommax_ipq50xx      | ✔      | ✔          | ✔          |
| qualcommax_ipq60xx      | ✔      | ✔          | ✔          |
| qualcommax_ipq807x      | ✔      | ✔          | ✔          |
| qualcommbe_ipq95xx      | ✔      | ✔          | ✔          |
| ramips_mt7620           | ✔      | ✘          |            |
| ramips_mt7621           | ✔      | ✘          |            |
| ramips_mt76x8           | ✔      | ✘          |            |
| ramips_rt288x           | ✔      | ✘          |            |
| ramips_rt305x           | ✔      | ✘          |            |
| ramips_rt3883           | ✔      | ✘          |            |
| realtek_rtl838x         | ✔      | ✘          |            |
| realtek_rtl839x         | ✔      | ✘          |            |
| realtek_rtl930x         | ✔      | ✘          |            |
| realtek_rtl930x_nand    | ✔      | ✘          |            |
| realtek_rtl931x         | ✔      | ✘          |            |
| realtek_rtl931x_nand    | ✔      | ✘          |            |
| rockchip_armv8          | ✔      | ✔          | ✔          |
| sifiveu_generic         |        | ✘          |            |
| siflower_sf19a2890      | ✔      | ✘          |            |
| siflower_sf21           |        | ✘          |            |
| stm32_stm32mp1          | ✔      | ✔          |            |
| sunxi_cortexa53         | ✔      | ✔          | ✔          |
| sunxi_cortexa7          | ✔      | ✔          |            |
| sunxi_cortexa8          | ✔      | ✔          |            |
| tegra_generic           | ✔      | ✔          |            |
| uml_generic             | ✔      | ✔          | ✔          |
| x86_64                  | ✔      | ✔          | ✔          |
| x86_generic             | ✔      | ✔          |            |
| x86_geode               | ✔      | ✔          |            |
| x86_legacy              | ✔      | ✔          |            |
| zynq_generic            | ✔      | ✔          |            |
+-------------------------+--------+------------+------------+
| ✔ = enabled                                                |
| ✘ = available but disabled                                 |
| blank = unavailable                                        |
+------------------------------------------------------------+

1. https://gist.github.com/graysky2/c77c9d063d4ade90f079e4fcd1b64080

Signed-off-by: John Audia <therealgraysky@proton.me>

graysky2 commented:
OK... build testing for x86/64-glibc fails to pull in libtcmalloc.so.4. I am tapped out for now. What do you think?
```
Package snort3 is missing dependencies for the following libraries:
libtcmalloc.so.4
```

BKPepe commented:
It looks like, you need to take your time, dig deeper inside the OpenWrt build system to know how to do it or you can try miss or hit, something will going to work. :) 

It is possible that the OpenWrt community on [IRC](https://openwrt.org/irc) will help you further.

graysky2 commented:
Plz see latest commit:

<details>
<summary>Click here</summary>

| Target_Subtarget | Snort3 | Gperftools | Vectorscan |
|------------------|--------|------------|------------|
| airoha_an7581    | ✔    | ✔        | ✔        |
| airoha_an7583    | ✔    | ✔        | ✔        |
| airoha_en7523    | ✔    | ✔        |            |
| apm821xx_nand    | ✔    |            |            |
| apm821xx_sata    | ✔    |            |            |
| at91_sam9x       | ✔    | ✔        |            |
| at91_sama5       | ✔    | ✔        |            |
| at91_sama7       | ✔    | ✔        |            |
| ath79_generic    | ✔    | ✔        |            |
| ath79_mikrotik   | ✔    | ✔        |            |
| ath79_nand       | ✔    | ✔        |            |
| ath79_tiny       | ✔    | ✔        |            |
| bcm27xx_bcm2708  | ✔    | ✔        |            |
| bcm27xx_bcm2709  | ✔    | ✔        |            |
| bcm27xx_bcm2710  | ✔    | ✔        | ✔        |
| bcm27xx_bcm2711  | ✔    | ✔        | ✔        |
| bcm27xx_bcm2712  | ✔    | ✔        | ✔        |
| bcm47xx_generic  | ✔    | ✔        |            |
| bcm47xx_legacy   | ✔    | ✔        |            |
| bcm47xx_mips74k  | ✔    | ✔        |            |
| bcm4908_generic  | ✔    | ✔        | ✔        |
| bcm53xx_generic  | ✔    | ✔        |            |
| bmips_bcm6318    | ✔    | ✔        |            |
| bmips_bcm63268   | ✔    | ✔        |            |
| bmips_bcm6328    | ✔    | ✔        |            |
| bmips_bcm6358    | ✔    | ✔        |            |
| bmips_bcm6362    | ✔    | ✔        |            |
| bmips_bcm6368    | ✔    | ✔        |            |
| d1_generic       |        | ✘        |            |
| econet_en751221  | ✔    | ✔        |            |
| gemini_generic   | ✔    | ✔        |            |
| imx_cortexa53    | ✔    | ✔        | ✔        |
| imx_cortexa7     | ✔    | ✔        |            |
| imx_cortexa9     | ✔    | ✔        |            |
| ipq40xx_chromium | ✔    | ✔        |            |
| ipq40xx_generic  | ✔    | ✔        |            |
| ipq40xx_mikrotik | ✔    | ✔        |            |
| ipq806x_chromium | ✔    | ✔        |            |
| ipq806x_generic  | ✔    | ✔        |            |
| ixp4xx_generic   | ✔    | ✔        |            |
| kirkwood_generic | ✔    | ✔        |            |
| lantiq_ase       | ✔    | ✔        |            |
| lantiq_falcon    | ✔    | ✔        |            |
| lantiq_xrx200    | ✔    | ✔        |            |
| lantiq_xrx200_legacy | ✔    | ✔        |            |
| lantiq_xway      | ✔    | ✔        |            |
| lantiq_xway_legacy | ✔    | ✔        |            |
| layerscape_armv7 | ✔    | ✔        |            |
| layerscape_armv8_64b | ✔    | ✔        | ✔        |
| loongarch64_generic |        | ✘        |            |
| malta_be         | ✔    | ✔        |            |
| malta_be64       | ✔    | ✔        |            |
| malta_le         | ✔    | ✔        |            |
| malta_le64       |        | ✘        |            |
| mediatek_filogic | ✔    | ✔        | ✔        |
| mediatek_mt7622  | ✔    | ✔        | ✔        |
| mediatek_mt7623  | ✔    | ✔        |            |
| mediatek_mt7629  | ✔    | ✔        |            |
| mpc85xx_p1010    | ✔    |            |            |
| mpc85xx_p1020    | ✔    |            |            |
| mpc85xx_p2020    | ✔    |            |            |
| mvebu_cortexa53  | ✔    | ✔        | ✔        |
| mvebu_cortexa72  | ✔    | ✔        | ✔        |
| mvebu_cortexa9   | ✔    | ✔        |            |
| mxs_generic      | ✔    | ✔        |            |
| omap_generic     | ✔    | ✔        |            |
| pistachio_generic | ✔    | ✔        |            |
| qoriq_generic    |        | ✘        |            |
| qualcommax_ipq50xx | ✔    | ✔        | ✔        |
| qualcommax_ipq60xx | ✔    | ✔        | ✔        |
| qualcommax_ipq807x | ✔    | ✔        | ✔        |
| qualcommbe_ipq95xx | ✔    | ✔        | ✔        |
| ramips_mt7620    | ✔    | ✔        |            |
| ramips_mt7621    | ✔    | ✔        |            |
| ramips_mt76x8    | ✔    | ✔        |            |
| ramips_rt288x    | ✔    | ✔        |            |
| ramips_rt305x    | ✔    | ✔        |            |
| ramips_rt3883    | ✔    | ✔        |            |
| realtek_rtl838x  | ✔    | ✔        |            |
| realtek_rtl839x  | ✔    | ✔        |            |
| realtek_rtl930x  | ✔    | ✔        |            |
| realtek_rtl930x_nand | ✔    | ✔        |            |
| realtek_rtl931x  | ✔    | ✔        |            |
| realtek_rtl931x_nand | ✔    | ✔        |            |
| rockchip_armv8   | ✔    | ✔        | ✔        |
| sifiveu_generic  |        | ✘        |            |
| siflower_sf19a2890 | ✔    | ✔        |            |
| siflower_sf21    |        | ✘        |            |
| stm32_stm32mp1   | ✔    | ✔        |            |
| sunxi_cortexa53  | ✔    | ✔        | ✔        |
| sunxi_cortexa7   | ✔    | ✔        |            |
| sunxi_cortexa8   | ✔    | ✔        |            |
| tegra_generic    | ✔    | ✔        |            |
| uml_generic      | ✔    | ✔        | ✘        |
| x86_64           | ✔    | ✔        | ✔        |
| x86_generic      | ✔    | ✔        |            |
| x86_geode        | ✔    | ✔        |            |
| x86_legacy       | ✔    | ✔        |            |
| zynq_generic     | ✔    | ✔        |            |

✔ means enabled ; ✘ means available but disabled ; blank means unavailable

</details>

BKPepe commented on net/snort3/Makefile:
```diff
@@ -21,15 +21,34 @@ PKG_CPE_ID:=cpe:/a:snort:snort
-    +PACKAGE_vectorscan-runtime:vectorscan-runtime
+    +PACKAGE_snort3&&@!(USE_MUSL&&(powerpc)):gperftools-runtime \
+    +PACKAGE_snort3&&TARGET_x86_64:vectorscan-runtime \
+    +PACKAGE_snort3&&aarch64:vectorscan-runtime
```
PACKAGE_snort3 is not necessary as this is already snort3.

BKPepe commented:
Has it been tested inside CI/CD?

graysky2 commented:
> Has it been tested inside CI/CD? 

No, not sure how to get it to run... perhaps you need to pull and push to the main packages

BKPepe commented:
I think now you need to rebase it and resolve conflict. Then, hopefully CI/CD will be triggered.

graysky2 commented:
I am not certain how to rebase it since I forked your fork and am PRing into that... shouldn't you rebase against openwrt/package?

BKPepe committed:
Try some changes in actions-shared-workflows

Signed-off-by: Josef Schlehofer <pepe.schlehofer@gmail.com>

BKPepe commented:
Heh, I rebased it first before asking. :) Because it was still not working, I rebased your branch. Now, its up-to-date. Don't forget to pull it locally before doing any changes. 

CI/CD is building master branch and I see that it is failing.

- aarch64:
```
2025-10-03T09:55:47.8466709Z snort version 3.9.5.0
2025-10-03T09:55:47.8466926Z 
2025-10-03T09:55:47.8467055Z Install options:
2025-10-03T09:55:47.8467375Z     prefix:     /usr
2025-10-03T09:55:47.8467725Z     includes:   /usr/include/snort
2025-10-03T09:55:47.8468127Z     plugins:    /usr/lib/snort
2025-10-03T09:55:47.8468369Z 
2025-10-03T09:55:47.8468686Z Compiler options:
2025-10-03T09:55:47.8469321Z     CC:             /builder/staging_dir/toolchain-aarch64_generic_gcc-14.3.0_musl/bin/aarch64-openwrt-linux-musl-gcc
2025-10-03T09:55:47.8470070Z     CXX:            /builder/staging_dir/toolchain-aarch64_generic_gcc-14.3.0_musl/bin/aarch64-openwrt-linux-musl-g++
2025-10-03T09:55:47.8472382Z     CFLAGS:         -Os -pipe -mcpu=generic -fno-caller-saves -fno-plt -fhonour-copts -ffile-prefix-map=/builder/build_dir/target-aarch64_generic_musl/snort3-3.9.5.0=snort3-3.9.5.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -I/builder/staging_dir/target-aarch64_generic_musl/usr/include/daq3 -I/builder/staging_dir/target-aarch64_generic_musl/usr/include/tirpc  -fvisibility=hidden   -DNDEBUG  -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free  -DNDEBUG
2025-10-03T09:55:47.8476483Z     CXXFLAGS:       -Os -pipe -mcpu=generic -fno-caller-saves -fno-plt -fhonour-copts -ffile-prefix-map=/builder/build_dir/target-aarch64_generic_musl/snort3-3.9.5.0=snort3-3.9.5.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -I/builder/staging_dir/target-aarch64_generic_musl/usr/include/daq3 -I/builder/staging_dir/target-aarch64_generic_musl/usr/include/tirpc  -fvisibility=hidden   -DNDEBUG  -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free  -DNDEBUG
2025-10-03T09:55:47.8479192Z     EXE_LDFLAGS:    -L/builder/staging_dir/toolchain-aarch64_generic_gcc-14.3.0_musl/usr/lib -L/builder/staging_dir/toolchain-aarch64_generic_gcc-14.3.0_musl/lib -fuse-ld=bfd -znow -zrelro -L/builder/staging_dir/target-aarch64_generic_musl/usr/lib/daq3 -ltirpc    
2025-10-03T09:55:47.8480997Z     MODULE_LDFLAGS: -L/builder/staging_dir/toolchain-aarch64_generic_gcc-14.3.0_musl/usr/lib -L/builder/staging_dir/toolchain-aarch64_generic_gcc-14.3.0_musl/lib -fuse-ld=bfd -znow -zrelro -L/builder/staging_dir/target-aarch64_generic_musl/usr/lib/daq3 -ltirpc -Wl,-Bsymbolic-functions    
2025-10-03T09:55:47.8481924Z 
2025-10-03T09:55:47.8481998Z Feature options:
2025-10-03T09:55:47.8482190Z     DAQ Modules:    Dynamic
2025-10-03T09:55:47.8482405Z     libatomic:      User-specified
2025-10-03T09:55:47.8482635Z     Hyperscan:      ON
2025-10-03T09:55:47.8482820Z     ICONV:          ON
2025-10-03T09:55:47.8483003Z     Libunwind:      OFF
2025-10-03T09:55:47.8483180Z     LZMA:           ON
2025-10-03T09:55:47.8483365Z     RPC DB:         TIRPC
2025-10-03T09:55:47.8483557Z     SafeC:          OFF
2025-10-03T09:55:47.8484000Z     TCMalloc:       ON
2025-10-03T09:55:47.8484219Z     JEMalloc:       OFF
2025-10-03T09:55:47.8484398Z     UUID:           ON
2025-10-03T09:55:47.8484575Z     NUMA:           OFF
2025-10-03T09:55:47.8484747Z     LibML:          OFF
2025-10-03T09:55:47.8484956Z -------------------------------------------------------
2025-10-03T09:55:47.8485137Z 
2025-10-03T09:55:47.8623352Z -- Configuring done (3.6s)
2025-10-03T09:55:48.1477875Z -- Generating done (0.3s)
2025-10-03T09:55:48.1486481Z -- Build files have been written to: /builder/build_dir/target-aarch64_generic_musl/snort3-3.9.5.0
2025-10-03T09:55:48.1786399Z touch /builder/build_dir/target-aarch64_generic_musl/snort3-3.9.5.0/.configured_68b329da9893e34099c7d8ad5cb9c940
2025-10-03T09:55:48.1831499Z rm -f /builder/build_dir/target-aarch64_generic_musl/snort3-3.9.5.0/.built
2025-10-03T09:55:48.1853931Z touch /builder/build_dir/target-aarch64_generic_musl/snort3-3.9.5.0/.built_check
2025-10-03T09:55:48.1898156Z MAKEFLAGS="--jobserver-auth=3,4" /builder/staging_dir/host/bin/ninja   -C /builder/build_dir/target-aarch64_generic_musl/snort3-3.9.5.0 
2025-10-03T09:55:48.1948421Z ninja: Entering directory `/builder/build_dir/target-aarch64_generic_musl/snort3-3.9.5.0'
2025-10-03T09:55:48.2209479Z ninja: error: '/builder/staging_dir/target-aarch64_generic_musl/usr/lib/libtcmalloc.so', needed by 'src/snort', missing and no known rule to make it
2025-10-03T09:55:48.2218483Z make[2]: *** [Makefile:122: /builder/build_dir/target-aarch64_generic_musl/snort3-3.9.5.0/.built] Error 1
2025-10-03T09:55:48.2219060Z make[2]: Leaving directory '/feed/net/snort3'
2025-10-03T09:55:48.2222565Z time: package/feeds/packages_ci/snort3/compile#1.94#2.71#4.34
2025-10-03T09:55:48.2231251Z     ERROR: package/feeds/packages_ci/snort3 failed to build.
2025-10-03T09:55:48.2236530Z make[1]: *** [package/Makefile:183: package/feeds/packages_ci/snort3/compile] Error 1
```

-  arm_cortex-a9_vfpv3-d16:

```
2025-10-03T09:46:16.5372605Z snort version 3.9.5.0
2025-10-03T09:46:16.5372817Z 
2025-10-03T09:46:16.5372945Z Install options:
2025-10-03T09:46:16.5373243Z     prefix:     /usr
2025-10-03T09:46:16.5373571Z     includes:   /usr/include/snort
2025-10-03T09:46:16.5373949Z     plugins:    /usr/lib/snort
2025-10-03T09:46:16.5374218Z 
2025-10-03T09:46:16.5374336Z Compiler options:
2025-10-03T09:46:16.5375108Z     CC:             /builder/staging_dir/toolchain-arm_cortex-a9+vfpv3-d16_gcc-14.3.0_musl_eabi/bin/arm-openwrt-linux-muslgnueabi-gcc
2025-10-03T09:46:16.5376388Z     CXX:            /builder/staging_dir/toolchain-arm_cortex-a9+vfpv3-d16_gcc-14.3.0_musl_eabi/bin/arm-openwrt-linux-muslgnueabi-g++
2025-10-03T09:46:16.5380099Z     CFLAGS:         -Os -pipe -fno-caller-saves -fno-plt -fhonour-copts -mfloat-abi=hard -ffile-prefix-map=/builder/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/snort3-3.9.5.0=snort3-3.9.5.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -I/builder/staging_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/usr/include/daq3 -I/builder/staging_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/usr/include/tirpc  -fvisibility=hidden   -DNDEBUG  -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free  -DNDEBUG
2025-10-03T09:46:16.5386860Z     CXXFLAGS:       -Os -pipe -fno-caller-saves -fno-plt -fhonour-copts -mfloat-abi=hard -ffile-prefix-map=/builder/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/snort3-3.9.5.0=snort3-3.9.5.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -I/builder/staging_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/usr/include/daq3 -I/builder/staging_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/usr/include/tirpc  -fvisibility=hidden   -DNDEBUG  -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free  -DNDEBUG
2025-10-03T09:46:16.5391966Z     EXE_LDFLAGS:    -L/builder/staging_dir/toolchain-arm_cortex-a9+vfpv3-d16_gcc-14.3.0_musl_eabi/usr/lib -L/builder/staging_dir/toolchain-arm_cortex-a9+vfpv3-d16_gcc-14.3.0_musl_eabi/lib -fuse-ld=bfd -znow -zrelro -L/builder/staging_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/usr/lib/daq3 -ltirpc    
2025-10-03T09:46:16.5395284Z     MODULE_LDFLAGS: -L/builder/staging_dir/toolchain-arm_cortex-a9+vfpv3-d16_gcc-14.3.0_musl_eabi/usr/lib -L/builder/staging_dir/toolchain-arm_cortex-a9+vfpv3-d16_gcc-14.3.0_musl_eabi/lib -fuse-ld=bfd -znow -zrelro -L/builder/staging_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/usr/lib/daq3 -ltirpc -Wl,-Bsymbolic-functions    
2025-10-03T09:46:16.5397098Z 
2025-10-03T09:46:16.5397221Z Feature options:
2025-10-03T09:46:16.5397534Z     DAQ Modules:    Dynamic
2025-10-03T09:46:16.5397893Z     libatomic:      User-specified
2025-10-03T09:46:16.5398278Z     Hyperscan:      OFF
2025-10-03T09:46:16.5398593Z     ICONV:          ON
2025-10-03T09:46:16.5398915Z     Libunwind:      OFF
2025-10-03T09:46:16.5399208Z     LZMA:           ON
2025-10-03T09:46:16.5399512Z     RPC DB:         TIRPC
2025-10-03T09:46:16.5399829Z     SafeC:          OFF
2025-10-03T09:46:16.5400132Z     TCMalloc:       ON
2025-10-03T09:46:16.5400432Z     JEMalloc:       OFF
2025-10-03T09:46:16.5400735Z     UUID:           ON
2025-10-03T09:46:16.5401184Z     NUMA:           OFF
2025-10-03T09:46:16.5401475Z     LibML:          OFF
2025-10-03T09:46:16.5401811Z -------------------------------------------------------
2025-10-03T09:46:16.5402120Z 
2025-10-03T09:46:16.5544791Z -- Configuring done (3.4s)
2025-10-03T09:46:16.8593163Z -- Generating done (0.3s)
2025-10-03T09:46:16.8603253Z -- Build files have been written to: /builder/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/snort3-3.9.5.0
2025-10-03T09:46:16.8975303Z touch /builder/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/snort3-3.9.5.0/.configured_68b329da9893e34099c7d8ad5cb9c940
2025-10-03T09:46:16.9021876Z rm -f /builder/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/snort3-3.9.5.0/.built
2025-10-03T09:46:16.9044533Z touch /builder/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/snort3-3.9.5.0/.built_check
2025-10-03T09:46:16.9089872Z MAKEFLAGS="--jobserver-auth=3,4" /builder/staging_dir/host/bin/ninja   -C /builder/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/snort3-3.9.5.0 
2025-10-03T09:46:16.9141327Z ninja: Entering directory `/builder/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/snort3-3.9.5.0'
2025-10-03T09:46:16.9414943Z ninja: error: '/builder/staging_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/usr/lib/libtcmalloc.so', needed by 'src/snort', missing and no known rule to make it
2025-10-03T09:46:16.9421802Z make[2]: *** [Makefile:122: /builder/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/snort3-3.9.5.0/.built] Error 1
```

graysky2 commented:
@BKPepe - can you rebase so the CI can run on this? In my testing locally, this change fixes the dependencies for the subtargets tested (see commit message for the table: https://github.com/BKPepe/packages/pull/4/commits/fcba3cb7ca4fc51814b05d31f3f09c7f7381b836)

BKPepe commented:
You need to rebase your branch on top of my branch. I can do that for you,
but not right now as I am not near the computer. I will do that in the late
afternoon.

Dne so 4. 10. 2025 13:23 uživatel graysky ***@***.***> napsal:

> *graysky2* left a comment (BKPepe/packages#4)
> <https://github.com/BKPepe/packages/pull/4#issuecomment-3368142124>
>
> @BKPepe <https://github.com/BKPepe> - can you rebase so the CI can run on
> this? In my testing locally, this change fixes the dependencies for the
> subtargets tested.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/BKPepe/packages/pull/4#issuecomment-3368142124>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA7IDVAYCTLYKWC327P7A6D3V6U3JAVCNFSM6AAAAACHVL6ULGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTGNRYGE2DEMJSGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

graysky2 commented:
I think I got it:
```
% git fetch BKP
% git rebase BKP/snort3-update
```

graysky2 commented:
Apparently I do not have a CircleCI pipeline configured so no checks are running. In contrast, when you did the rebase last time, the checks I see when I send a PR against the official project run.

BKPepe commented:
It's running now.

graysky2 commented:
I see that... did this just happen or did you have to manually intervene?

BKPepe commented:
I just need to approve to run the workflow as I do have configured in the repository settings. Almost like in the OpenWrt packages. For new contributors to the repo and as well for newly created accounts, who submitted PR(s), you need to manually press the button to avoid using resources for bad habbits.

graysky2 commented:
@BKPepe - please see latest commit, and if you can kick the CI that would be great.
